NEWS.md

```diff
@@ -2,6 +2,12 @@
 
 ### Bug Fixes
 
+- Previously, when the tape optimizer from `TMB` was switched on a warning would be given, instructing to turn off the tape optimizer. However, this is not necessary for reproducible results. Instead now it is checked whether the deterministic hash for the TMB tape optimizer is used, and a warning is issued otherwise.
```

Comment:
Might make sense to say that this resulted in the first fit being fit with a different optimizer and thus being different?